--- a/Makefile	2017-06-09 16:52:23.347638947 +0200
+++ b/Makefile	2017-06-09 16:52:36.791757513 +0200
@@ -1,10 +1,23 @@
-ifneq ($(KERNELRELEASE),)
+#ifneq ($(KERNELRELEASE),)
 
 # Uncomment the following line to enable kernel stack unwinding within gator, or update gator_backtrace.c
 # EXTRA_CFLAGS +=   -DGATOR_KERNEL_STACK_UNWINDING
-
+TARGET = gator
 CONFIG_GATOR ?= m
-obj-$(CONFIG_GATOR) := gator.o
+
+$(TARGET)-objs := gator_main.o \
+	gator_events_block.o \
+	gator_events_irq.o \
+	gator_events_meminfo.o \
+	gator_events_mmapped.o \
+	gator_events_net.o \
+	gator_events_sched.o 
+
+obj-m := $(TARGET).o
+
+SRC := $(shell pwd)
+
+
 
 gator-y :=  gator_main.o \
             gator_events_block.o \
@@ -67,14 +80,14 @@
 endif
 
 # Should the original or new block_rq_complete API be used?
-OLD_BLOCK_RQ_COMPLETE := $(shell grep -A3 block_rq_complete $(srctree)/include/trace/events/block.h | grep nr_bytes -q; echo $$?)
+OLD_BLOCK_RQ_COMPLETE := $(shell grep -A3 block_rq_complete $(KERNEL_SRC)/include/trace/events/block.h | grep nr_bytes -q; echo $$?)
 EXTRA_CFLAGS += -DOLD_BLOCK_RQ_COMPLETE=$(OLD_BLOCK_RQ_COMPLETE)
 
 gator-$(CONFIG_ARM) +=  gator_events_l2c-310.o
 
 gator-$(CONFIG_ARM64) +=
 
-$(obj)/gator_main.o: $(obj)/gator_src_md5.h
+#gator_main.o: gator_src_md5.h
 
 clean-files := gator_src_md5.h
 
@@ -86,21 +99,27 @@
        chk_events.h = :
  quiet_chk_events.h = echo '  CHK     $@'
 silent_chk_events.h = :
-$(obj)/gator_src_md5.h: FORCE
-	@$($(quiet)chk_events.h)
-	$(Q)cd $(srctree) && cd $(src) ; $(CONFIG_SHELL) -c "echo 'static char *gator_src_md5 = \"'\`ls *.c *.h mali/*.h | grep -Ev '^(gator_src_md5\.h|gator\.mod\.c)$$' | LC_ALL=C sort | xargs cat | md5sum | cut -b 1-32\`'\";'" > $(abspath $@)
-
-else
+#gator_src_md5.h:
+#	@$($(quiet)chk_events.h)
+#	cd $(SRC) ; /bin/bash -c "echo 'static char *gator_src_md5 = \"'\`ls *.c *.h mali/*.h | grep -Ev '^(gator_src_md5\.h|gator\.mod\.c)$$' | LC_ALL=C sort | xargs cat | md5sum | cut -b 1-32\`'\";'" > $(abspath $@)
+
+#else
+
+#all:
+#	@echo
+#	@echo "usage:"
+#	@echo "      make -C <kernel_build_dir> M=\`pwd\` ARCH=arm CROSS_COMPILE=<...> modules"
+#	@echo
+#	$(error)
 
 all:
-	@echo
-	@echo "usage:"
-	@echo "      make -C <kernel_build_dir> M=\`pwd\` ARCH=arm CROSS_COMPILE=<...> modules"
-	@echo
-	$(error)
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD)
+
+modules_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) modules_install
 
 clean:
 	rm -f *.o .*.cmd gator_src_md5.h modules.order Module.symvers gator.ko gator.mod.c
 	rm -rf .tmp_versions
 
-endif
+#endif
